vyper/cli/vyper_compile.py

```diff
@@ -143,65 +154,89 @@ def exc_handler(contract_name: ContractName, exception: Exception) -> None:
     raise exception
 
 
-def get_interface_codes(codes: ContractCodes) -> Any:
-    interface_codes = {}
-    interfaces = {}
+def get_interface_codes(root_path: Path, codes: ContractCodes) -> Any:
+    interface_codes: Dict = {}
+    interfaces: Dict = {}
 
-    for code in codes.values():
-        interface_codes.update(extract_file_interface_imports(code))
+    for file_path, code in codes.items():
+        interfaces[file_path] = {}
+        parent_path = root_path.joinpath(file_path).parent
 
-    if interface_codes:
+        interface_codes = extract_file_interface_imports(code)
         for interface_name, interface_path in interface_codes.items():
-            file_path = Path(interface_path.replace('.', '/')).resolve()
 
-            try:
-                suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists())
-            except StopIteration:
-                raise Exception(
-                    f'Imported interface "{interface_path}{{.vy,.json}}" does not exist.'
+            base_paths = [parent_path]
+            if not interface_path.startswith('.'):
+                base_paths.append(root_path)
+            elif interface_path.startswith('../') and parent_path == root_path:
+                raise FileNotFoundError(
+                    f"{file_path} - Cannot perform relative import outside of base folder"
                 )
 
-            valid_path = file_path.with_suffix(suffix)
+            valid_path = get_interface_file_path(base_paths, interface_path)
             with valid_path.open() as fh:
                 code = fh.read()
                 if valid_path.suffix == '.json':
-                    interfaces[interface_name] = {
+                    interfaces[file_path][interface_name] = {
                         'type': 'json',
                         'code': json.loads(code.encode())
                     }
                 else:
-                    interfaces[interface_name] = {
+                    interfaces[file_path][interface_name] = {
                         'type': 'vyper',
                         'code': code
                     }
 
     return interfaces
 
 
+def get_interface_file_path(base_paths: Sequence, import_path: str) -> Path:
+    relative_path = Path(import_path)
+    for path in base_paths:
+        file_path = path.joinpath(relative_path)
+        try:
+            suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists())
+            return file_path.with_suffix(suffix)
+        except StopIteration:
+            continue
```

Comment:
Not a necessary change - just something interesting I found: Apparently you can now pass None in to next to make the control flow nicer:
```python
suffix = next(i for i in ('.vy', '.json') if file_path.with_suffix(i).exists(), None)
if suffix:
   return file_path.with_suffix(suffix)

Reply:
:+1:  it looks cleaner without the try/catch, will add